### DISCUSSION TITLE
Is RS485 for a chlorinator even worth it?

### DISCUSSION THREAD

**Discussion by jamestalmage** [tagyoureit/nodejs-poolController#729] [Q&A]
Reading the Intellichlor install manual, it seems choosing chlorinator levels would be pretty simple with just a relay:

> • 0% - No LEDs lit - No chlorine produced - IntelliChlor Electronic Chlorine Generator is off.
• 20% - 1 LED lit - Produces chlorine 20% of each hour of pump run time, 12 minutes on, 48 minutes off.
• 40% - 2 LEDs lit - Produces chlorine 40% of each hour of pump run time, 24 minutes on, 36 minutes off.
• 60% - 3 LEDs lit - Produces chlorine 60% of each hour of pump run time, 36 minutes on, 24 minutes off.
• 80% - 4 LEDs lit - Produces chlorine 80% of each hour of pump run time, 48 minutes on, 12 minutes off.
• 100% - 5 LEDs lit - Produces chlorine almost 100% of each the hour of pump run time, 59 minutes on, 1 minute off. 

Seems like this would be simple to implement with any chlorinator and a relay... Just cycle power as described above. Is this implemented already? (If not, would a PR be welcome?)

Looking at the [chlorinator protocol](https://github.com/tagyoureit/nodejs-poolController/wiki/Chlorinator), it does seem like salt PPM is returned by some units... Is that information returned by all of them? I'm assuming it's well integrated into the dashPanel at this point?

**Comment by rstrouse** [comment]
It is nice to have the ability to adjust it as needed.   The other thing that these controllers do is reverse the polarity to reduce buildup so switching it off in a cycle will interfere with this function.  You are limited to Aquarite and Pentair.  However I do recommend Pentair. There was a period of time when everyone was carrying a flag for Circupool but that has waned due to issues with them.

**Comment by jamestalmage** [comment]
Getting further in the manual... It also talks about a mode to check the hours on the cell from the controller... Is that available over RS485?

Also, do you know... Is salt PPM returned by all intellichlor units?

**Reply by rstrouse** [reply]
> Getting further in the manual... It also talks about a mode to check the hours on the cell from the controller... Is that available over RS485?

This is not available on the RS485 comms.  The chlorinator controller only outputs the current setpoint, salt level, and status.  The status includes some things that are not indicated on the cell display or shown in a Pentair OCP.
* Low Flow
* Very Low Salt
* Low Salt
* Low Voltage
* High Current
* Low Water Temp
* Clean Cell

> Also, do you know... Is salt PPM returned by all intellichlor units?

Yes

**Reply by zombielinux** [reply]
@rstrouse could you clarify how to extract those additional values from an intellichlor's status message? 

I could certainly use "Low Flow" in some of my automation logic.

**Reply by tagyoureit** [reply]
The byte map is here: https://github.com/tagyoureit/nodejs-poolController/blob/a595fe061192e2d4b79038c923015791410a8093/controller/boards/SystemBoard.ts#L522-L532

And the byte itself is part of the Action: 18 message.  https://github.com/tagyoureit/nodejs-poolController/blob/a595fe061192e2d4b79038c923015791410a8093/controller/comms/messages/status/ChlorinatorStateMessage.ts#L120
